spec/base_spec.rb

```diff
@@ -647,6 +651,126 @@ def self.index_name
         end
       end
 
+      context "with a search query" do
+        subject(:response) { DummyElasticSearchModel.where(match_conditions.merge(_q: query_conditions)) }
+        let(:match_conditions) { {} }
+
+        context "string searching" do
+          before(:each) do
+            @dummy_model1 = DummyElasticSearchModel.create!(my_string: "Hello")
+            @dummy_model2 = DummyElasticSearchModel.create!(my_string: "Hello2")
+            @dummy_model3 = DummyElasticSearchModel.create!(my_string: "Hey1", my_other_string: "Hello3")
+            @dummy_model4 = DummyElasticSearchModel.create!(my_string: "Hey2", my_other_string: "Hello4")
+            refresh_index
+          end
+
+          context "alone" do
+            let(:query_conditions) { "Hello" }
+
+            it "returns all matching entries" do
+              expect(response.models.count).to eq(4)
+              expect_response_models_match(response.models, [@dummy_model1, @dummy_model2, @dummy_model3, @dummy_model4])
```

Comment:
Take with a grain of salt considering I just learned of the syntax, couldn't this pattern throughout all these tests become a shared spec in `shared_examples`?
```
expect_response_models_match(response.models, expected_models)
```

Reply:
I went back and forth on this..I decided against doing it that way because it didn't seem like it fit. 

It would DRY it out slightly, but not sure if its the best use of shared examples